packages/theme-a11y/theme-a11y.js

```diff
@@ -199,3 +199,127 @@ export function removeTrapFocus() {
   document.removeEventListener('focusout', trapFocusHandlers.focusout);
   document.removeEventListener('keydown', trapFocusHandlers.keydown);
 }
+
+/**
+ * Add a preventive message to external links and links that open to a new window.
+ * @param {object} messages - Custom messages to overwrite with keys: newWindow, external, newWindowExternal
+ * @param {string} messages.newWindow - When the link opens in a new window (e.g. target="_blank")
+ * @param {string} messages.external - When the link is external (e.g. www.cnn.com)
+ * @param {string} messages.newWindowExternal - When the link is external and opens in a new window (<a href="http://www.shopify.com" target="_blank">Shopify</a>)
+ * @param {object} targetLinks - Specific elements to be targeted
+ * @param {object} prefix - Prefix to namespace "id" of the messages
+ */
+export function accessibleLinks(messages, targetLinks, prefix) {
+  var accessibleLinksHandlers = {};
+  var messagesSource = messages || {};
+  var body = document.querySelector('body');
+  var prefixName = prefix || 'a11y';
+  var idSelectors = {
+    newWindow: prefixName + '-new-window-message',
+    external: prefixName + '-external-message',
+    newWindowExternal: prefixName + '-new-window-external-message'
+  };
+
+  if (targetLinks === undefined || targetLinks === null) {
+    targetLinks = document.querySelectorAll('a[href]:not([aria-describedby])');
+  }
+
+  if (targetLinks !== '' && typeof targetLinks !== 'object') {
+    throw new TypeError(targetLinks + ' is not an Element object.');
+  }
+
+  accessibleLinksHandlers.generateHTML = function(customMessages) {
+    if (typeof customMessages !== 'object') {
+      customMessages = {};
+    }
+
+    var localMessages = Object.assign(
+      {
+        newWindow: 'Opens in a new window.',
+        external: 'Opens external website.',
+        newWindowExternal: 'Opens external website in a new window.'
+      },
+      customMessages
+    );
+
+    var container = document.createElement('ul');
+    var htmlMessages = '';
```

Comment:
```suggestion
    var htmlMessages = Object.keys(messages).reduce(function(html, key){
      html += 
        '<li id=' +
        idSelectors[key] + 
        '>' +
        localMessages[key] +
        '</li>';
    }, '');
```

Reply:
This is solid, thank you! I always forgot `map`, `filter` and `reduce` being supported on IE9+